### DISCUSSION TITLE
replacing all images in a folder, with each image sized into a "frame"

### DISCUSSION THREAD

**Discussion by CuzzinChizzy** [ImageMagick/ImageMagick#5726] [Help]
Hello!

I have folders of game art images like these

![cat](https://user-images.githubusercontent.com/117551031/200149918-74d544a1-93dd-4b33-a30a-7e50d9f13fef.png)      ![pac](https://user-images.githubusercontent.com/117551031/200149923-dfce283e-9a40-4c56-833d-983eb5956fcb.png)

For the sake of uniformity I want to add a frame to every one of these images.

This frame

![frame](https://user-images.githubusercontent.com/117551031/200149935-d7ae2936-80a0-4405-893a-a6d93fa562f1.png)

for this result
![output](https://user-images.githubusercontent.com/117551031/200149998-c158358c-ed0f-4705-a4d1-1c7127f93f2a.png)

Using this string I was able to do what I want to do for one picture at a time

`magick composite -geometry 700x890!+49+140 cat.png  frame.png  output.png`

but with this I'd have to change the name of the overlay image for each game, and then also the output image.
I've been searching and experimenting for hours and can't for the life of me figure out how to do the whole batch. 

What would be the proper code to tell ImageMagick to overlay/fit each "game art".png in the folder on top of/into the frame image, and then to update each image once it's modified?

something along the lines of 
`magick composite -geometry 700x890!+49+140 (*).png  frame.png  *.png` (but with proper code)

I've tried a bunch different combinations of Mogrify and composite, and nothing is working. 

Any tips would be greatly appreciated!

**Comment by snibgo** [comment]
I suggest putting your command in a shell "for" loop.

**Reply by CuzzinChizzy** [reply]
thanks for the suggestion snibgo!

**Comment by fmw42** [comment]
Have you tried

`magick mogrify -draw 'image over 0,0 0,0 "path_to/frame.png"' *.png`

But since frame is a PNG and you are processing PNG images, you should not have the frame in the same directory or it will get processed also.

See https://imagemagick.org/Usage/basics/#mogrify_compose

**Reply by CuzzinChizzy** [reply]
ooh, I'm gonna try this mogrify with -draw command

and you're right about the frame needing to be a different format than the game art images!
I just switched the frame to a .jpg and left all the art as .png

**Comment by CuzzinChizzy** [comment]
After snibgo's suggestion I did some more digging and came across a few scripts with the _for loop_

I copied parts of em and just kept testing and making adjustments until I came up with this:

`For /R %f in (*.png) DO composite -geometry 700x890!+49+140 "%f" frame.jpg "%f"`


Did exactly what I wanted it to do!
This is going to be such a huge time saver for me, as there are a ton of other similar projects I want to do with thousands of pictures each. 

I'm still not exactly sure I completely understand what I'm saying in the script.
Guessing it's basically saying "**For** all .png files in the current directory, **DO** adjust the size and position, then  composite it on top of the frame.jpg, and output as the same named .png". (thus overwriting the old file with the adjusted one)

I'm guessing %f acts as a variable, that gets set to %f= "current filename in the batch queue".png
but what is /R?   return?    seems like it should be at the end of the line...  like "FOR this file, DO this action, then RETURN to loop start for next file"
But I'm a noob in scripting.

How is this different from something like this?
```
for f in *.png
  do   composite -geometry 700x890!+49+140 $f frame.jpg $f
  done
```

Anyways, I'll keep reading and learning more. Get my fluency up. 

Thanks again guys, this rules!

**Reply by fmw42** [reply]
That is Unix (Linux or Mac OSX) syntax, not Windows syntax.

**Comment by snibgo** [comment]
What is your shell? Windows CMD, or bash, or what?

"for ... in ... do ..." is Windows CMD syntax. In Windows CMD, type "help for" for help. "for /R" recursively searches subdirectories, rather than just the current directory.

"for ... in ... do ... done" is bash syntax. In bash, type "man bash" for help.

**Reply by CuzzinChizzy** [reply]
just windows command

thanks for the heads up on "help"
